### PR TITLE
Remove generics from key builder

### LINKED ISSUE #1027: Remove generics in key_builder
The [Key_builder](https://github.com/tigrisdata/tigris/blob/main/query/filter/key_builder.go) supports generics to allow `schema.Field` or `schema.QueryableField`. This should be removed and `key_builder` should only accept `schema.QueryableField`. 

Steps to get this working:
1. Remove generics in key_builder
2. Fix tests to only accept `schema.QueryableField` this could be done by converting the current `schema.Field` options into a `queryableField` before running the test
3. Add a method on `schema.DefaultCollection` to get the primarykeys as a list of queryablefields

### PR DESCRIPTION
**Pull request by karuppiah7890**
## Describe your changes

Remove generics from key builder

## How best to test these changes

`make test`

## Issue ticket number and link

Fixes #1027 

Related Discord discussion thread - https://discord.com/channels/1033842669983633488/1033843208192524348/1095960587806183494

### CODE REVIEW

**Comment by karuppiah7890** [comment]
I have done some changes with 0 knowledge of fields, queryable fields and the different entities in key builder source file. 

TODO:
- [ ] Rebase with latest `main`
- [ ] Check if all the tests pass with `make test`
   - [ ] `TestKeyBuilderStrictEq` test failed. Check it
   - [ ] Noticed some tests failing even before the PR changes intermittently and some after the PR changes. Check if it's PR related tests and check if it's flaky
- [ ] Understand if the changes in `base_runner.go` and `secondary_index_reader.go` make sense - converting fields to queryable fields that is
- [ ] Understand fields, queryable fields, key builder in detail

**Comment by ovaistariq** [comment]
Thank you for your contribution. I am looking forward to a finished version of it.

File: query/filter/key_builder.go [hunk 1]
```diff
@@ -124,7 +119,7 @@ func NewKeyBuilder[F fieldable](composer KeyComposer[F], primaryKey bool) *KeyBu
 // On each level multiple keys can be formed because the user can specify ranges. The builder is not deciding the logic
 // of key generation, the builder is simply traversing on the filters and calling compose where the logic resides.
 // If the build is for a primary key, the query plans are merged into a single plan.
-func (k *KeyBuilder[F]) Build(filters []Filter, userDefinedKeys []F) ([]QueryPlan, error) {
+func (k *KeyBuilder) Build(filters []Filter, userDefinedKeys []*schema.QueryableField) ([]QueryPlan, error) {
```
Comment (garrensmith) [inline query/filter/key_builder.go:122]: Lets rename `userDefinedKey` to indexedKeys this will make more sense here now.

**Review by garrensmith — CHANGES_REQUESTED** [review]
This is a really good start. I've left the next steps you need to do on the changes here. 

Some context:
1. Key build is used to take a set of fields defined on a document schema and query foundationdb for a document. 
2. It can accept an array of primary key fields to fetch against the primary index where the actual documents are stored.
3. Or it can be used to query against the secondary indexes.

File: query/filter/key_builder.go (hunk 1 above)
Comment (karuppiah7890) [reply query/filter/key_builder.go:122]: `Build` calls `Compose`, should we change all parameters named `userDefinedKeys` to `indexedKeys`? Does that make sense?

Comment (garrensmith) [reply query/filter/key_builder.go:122]: Yeah. I'm still not sure that is the best name. Hold off on that until last and we can decide.

File: server/services/v1/database/base_runner.go [hunk 2]
```diff
@@ -211,13 +211,17 @@ func (runner *BaseQueryRunner) buildKeysUsingFilter(coll *schema.DefaultCollecti
...
 	})
-	queryPlan, err := kb.Build(filters, coll.GetPrimaryKey().Fields)
+	queryPlan, err := kb.Build(filters, fieldsToQueryableFields(coll.GetPrimaryKey().Fields))
 	if err != nil {
 		return nil, err
 	}
 	return queryPlan[0].Keys, nil
 }
 
+func fieldsToQueryableFields(fields []*schema.Field) []*schema.QueryableField {
```
[the remark this answers is not in the document]
Comment (karuppiah7890) [reply server/services/v1/database/base_runner.go:221]: Thanks for explaining! That gives more context to understand better. I'm still reading the code to understand what's going on and the internal details. I think I should use the Tigris Web Console and API to understand better from a user perspective

**Review by garrensmith — APPROVED** [review]

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**